sdk/core/azure-core/tests/test_streaming.py

```diff
@@ -92,6 +92,22 @@ def test_decompress_compressed_no_header(http_request):
     except UnicodeDecodeError:
         pass
 
+@pytest.mark.parametrize("http_request", HTTP_REQUESTS)
+def test_compress_compressed_no_header_offline(port, http_request):
+    # expect compressed text
+    client = PipelineClient("")
+    try:
+        request = http_request(method="GET", url="http://localhost:{}/streams/compressed_no_header".format(port))
+        pipeline_response = client._pipeline.run(request, stream=True)
+        response = pipeline_response.http_response
+        data = response.stream_download(client._pipeline, decompress=False)
+        content = b"".join(list(data))
+        with pytest.raises(UnicodeDecodeError):
+            decoded = content.decode('utf-8')
+    except HttpResponseError as e:
+        print(e.response.text())
```

Comment:
we shouldn't be catching the HttpResponseError because we're not expecting the service to throw an error, so here we're swallowing if we do get a network error.

Is the current code getting a network error?